plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -115,17 +124,13 @@ public class IcebergSplitSource
     private final Domain pathDomain;
     private final Domain fileModifiedTimeDomain;
     private final OptionalLong limit;
-
+    private final AsyncQueue<FileScanTask> queue;
+    private volatile TrinoException taskGenerationException;
     private TupleDomain<IcebergColumnHandle> pushedDownDynamicFilterPredicate;
-    private CloseableIterable<FileScanTask> fileScanIterable;
-    private long targetSplitSize;
-    private CloseableIterator<FileScanTask> fileScanIterator;
-    private Iterator<FileScanTask> fileTasksIterator = emptyIterator();
-    private boolean fileHasAnyDeletions;
-
     private final boolean recordScannedFiles;
     private final ImmutableSet.Builder<DataFileWithDeleteFiles> scannedFiles = ImmutableSet.builder();
     private long outputRowsLowerBound;
+    private boolean enoughSplits;
```

Comment:
what is "enoughSplits"? the name hint at the  semantics to me

Reply:
Fair point. I have added a comment to describe its meaning. The main use of this flag is if there is a LIMIT clause: it indicates that no more splits are required for the scan, since we've already reached the required number of records to satisfy the LIMIT.

Reply:
I have removed `enoughSplits` actually. `FileAwareScanTask` will help enable the limit pushdown so this flag is no longer needed